stdlib/SparseArrays/src/linalg.jl

```diff
@@ -50,9 +50,9 @@ function mul!(C::StridedVecOrMat, A::SparseMatrixCSC, B::StridedVecOrMat, α::Nu
     C
 end
 *(A::SparseMatrixCSC{TA,S}, x::StridedVector{Tx}) where {TA,S,Tx} =
-    (T = promote_type(TA, Tx); mul!(similar(x, T, A.m), A, x, one(T), zero(T)))
+    (T = promote_op((a,b)->a*b+a*b, TA, Tx); mul!(similar(x, T, A.m), A, x, one(T), zero(T)))
```

Comment:
Cleaner to do `typeof(zero(TA)*zero(Tx)+zero(TA)*zero(Tx))`

Reply:
Actually, we already have a function for this in LinearAlgebra.  Do `promote_op(matprod, TA, tx)`, where you do `using LinearAlgebra: matprod` somewhere.

Reply:
Done, thanks!